### ISSUE TITLE
If VAULT_TOKEN is present, it attempts to connect to localhost:8200?

### ISSUE THREAD

**Issue by JoeOrtiz** [hashicorp/envconsul#111]
I can't always reproduce it somehow, but it appears that is the environment variable is present, even though no vault config is set in the config file, it attempts to:

```
Jun 23 17:16:21 i-b7f4f86d.node.us-west-2-tools bash[19989]: 2016/06/23 17:16:21 [ERR] (view) vault_token error renewing vault token: Put https://127.0.0.1:8200/v1/auth/token/renew-self: dial tcp 127.0.0.1:8200: getsockopt: conne
Jun 23 17:16:21 i-b7f4f86d.node.us-west-2-tools bash[19989]: 2016/06/23 17:16:21 [ERR] (runner) watcher reported error: error renewing vault token: Put https://127.0.0.1:8200/v1/auth/token/renew-self: dial tcp 127.0.0.1:8200: get
Jun 23 17:16:21 i-b7f4f86d.node.us-west-2-tools bash[19989]: 2016/06/23 17:16:21 [ERR] error renewing vault token: Put https://127.0.0.1:8200/v1/auth/token/renew-self: dial tcp 127.0.0.1:8200: getsockopt: connection refused
```

The config has no mention of vault, but attempts to connect ?

**Comment by sethvargo** [comment]
Hi @JoeOrtiz 

This is the correct behavior. If you specify a token, CT will attempt to renew that token for you.

**Comment by JoeOrtiz** [comment]
But... I think this should only be attempted if you have in your config defined a vault address, what if we want to pass it to consul-template, not envconsul ?

This is causing some conflicts where we just want to use envconsul to talk to just consul, but consul-template to vault, and pass that variable, and due to systemd preloads all the EnvironmentFile= data, it causes the above problem.

**Comment by eedwards-sk** [comment]
@sethvargo This is breaking me the same as JoeOrtiz. Can we please re-open?

In my scenario, VAULT_TOKEN is specified by a wrapper, just as you suggested in https://github.com/hashicorp/consul-template/issues/318#issuecomment-169731445

The vault server is not local and I'm providing the address in the config file, so why try to reach localhost and ignore the value I specified?

**Comment by jefferai** [comment]
It's probably also configuring VAULT_ADDRESS and overriding your config.

**Comment by eedwards-sk** [comment]
Just for posterity, once I turned off the renew setting for vault, it surfaced that it was failing to read the consul pki files:

```
2019/01/23 23:13:10.155855 [ERR] (cli) runner: runner: client set: consul: open /opt/snapkitchen/consul-pki/client-key.pem: permission denied
```

Then I removed the whole consul block, trying just vault, and turning back on renew, it worked fine!

So, there's definitely some strange behavior around how config loading works. In this case, having the vault renew token behavior enabled caused it to swallow the consul error, and then spam localhost trying to renew the token. Fascinating!
